lib/Sema/TypeCheckType.cpp

```diff
@@ -3756,160 +3756,6 @@ Type TypeChecker::substMemberTypeWithBase(ModuleDecl *module,
   return resultType;
 }
 
-namespace {
-
-class UnsupportedProtocolVisitor
-  : public TypeReprVisitor<UnsupportedProtocolVisitor>, public ASTWalker
-{
-  ASTContext &Ctx;
-  bool checkStatements;
-  bool hitTopStmt;
-    
-public:
-  UnsupportedProtocolVisitor(ASTContext &ctx, bool checkStatements)
-    : Ctx(ctx), checkStatements(checkStatements), hitTopStmt(false) { }
-
-  bool walkToTypeReprPre(TypeRepr *T) override {
-    if (T->isInvalid())
-      return false;
-    if (auto compound = dyn_cast<CompoundIdentTypeRepr>(T)) {
-      // Only visit the last component to check, because nested typealiases in
-      // existentials are okay.
-      visit(compound->getComponentRange().back());
-      return false;
-    }
-    // Arbitrary protocol constraints are OK on opaque types.
-    if (isa<OpaqueReturnTypeRepr>(T))
-      return false;
-    
-    visit(T);
-    return true;
-  }
-
-  std::pair<bool, Stmt*> walkToStmtPre(Stmt *S) override {
-    if (checkStatements && !hitTopStmt) {
-      hitTopStmt = true;
-      return { true, S };
-    }
-
-    return { false, S };
-  }
-
-  bool walkToDeclPre(Decl *D) override {
-    return !checkStatements;
-  }
-
-  void visitTypeRepr(TypeRepr *T) {
-    // Do nothing for all TypeReprs except the ones listed below.
-  }
-
-  void visitIdentTypeRepr(IdentTypeRepr *T) {
-    if (T->isInvalid())
-      return;
-    
-    auto comp = T->getComponentRange().back();
-    if (auto *proto = dyn_cast_or_null<ProtocolDecl>(comp->getBoundDecl())) {
-      if (!proto->existentialTypeSupported()) {
-        Ctx.Diags.diagnose(comp->getNameLoc(),
-                           diag::unsupported_existential_type,
-                           proto->getName());
-        T->setInvalid();
-      }
-    } else if (auto *alias = dyn_cast_or_null<TypeAliasDecl>(comp->getBoundDecl())) {
-      auto type = Type(alias->getDeclaredInterfaceType()->getDesugaredType());
-      type.findIf([&](Type type) -> bool {
-        if (T->isInvalid())
-          return false;
-        if (type->isExistentialType()) {
-          auto layout = type->getExistentialLayout();
-          for (auto *proto : layout.getProtocols()) {
-            auto *protoDecl = proto->getDecl();
-
-            if (protoDecl->existentialTypeSupported())
```

Comment:
I think existentialTypeSupported() is dead now too

Reply:
Nice! It seems to be used in ExistentialConformsToSelfRequest and one other place in type checker. Let me try removing it and see if anything breaks.

Reply:
Since only @objc protocols can self-conform, we shouldn't need it there since @objc protocols already cannot contain associated types or requirements with Self in non-return position.